### PR TITLE
Envdir support

### PR DESCRIPTION
**Pull request by tsibley**
See commit messages.

Envdir support is useful in the general case, but specifically added now to enable [better facilities for environment variables in Nextstrain CLI](https://github.com/tsibley/blab-standup/blob/master/2023-06-02.md).

### Testing

- [x] Locally with various permutations
- [x] Checks pass

### CODE REVIEW

**Comment by tsibley** [comment]
Note that the `NEXTSTRAIN_ENVD_URL` tests are skipped in CI right now ([e.g.](https://github.com/nextstrain/docker-base/actions/runs/5205904685/jobs/9392876826?pr=158#step:7:15)) because there's no AWS credentials configured.

```
cram -v tests/
tests/basic.t: passed
tests/envdir-url.t: skipped
tests/envdir.t: passed
# Ran 3 tests, 1 skipped, 0 failed.
```

**Review by victorlin — APPROVED** [review]
Browsed through and ran envdir.t locally. Did not test envdir-url.t.

File: tests/envdir.t [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env cram
+
+Setup.
+
+  $ : "${IMAGE:=localhost:5000/nextstrain/base:latest}"
+  $ (docker image inspect "$IMAGE" || docker image pull "$IMAGE") &>/dev/null
+
+  $ envd="$(mktemp -d "$TMP"/env.d-XXXXX)"
```
[the remark this answers is not in the document]
Comment (victorlin) [reply tests/envdir.t:8]: You could also use the default working directory, which is a directory within `$TMP` that is unique per test.

```suggestion
  $ envd=env.d
```

[the remark this answers is not in the document]
Comment (tsibley) [reply tests/envdir.t:8]: OH! Yes, that's simpler and nicer. Will do.

[the remark this answers is not in the document]
Comment (tsibley) [reply tests/envdir.t]: Done in the repush.

**Comment by tsibley** [comment]
Woo, the GitHub/AWS OIDC integration worked a treat ([c.f.](https://github.com/nextstrain/docker-base/actions/runs/5258273874/jobs/9504085048?pr=158#step:9:20)).

```
cram -v tests/
tests/basic.t: passed
tests/envdir-url.t: passed
tests/envdir.t: passed
# Ran 3 tests, 0 skipped, 0 failed.
```

**Pull request merged**